src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -38,350 +39,52 @@ namespace triton { namespace perfanalyzer {
 
 /// Class to test the RequestRateManager
 ///
-class TestRequestRateManager : public RequestRateManager {
+class TestRequestRateManager : public TestLoadManagerBase,
+                               public RequestRateManager {
  public:
-  TestRequestRateManager()
+  ~TestRequestRateManager() = default;
```

Comment:
There's no need to declare/define a destructor here. It will be implicitly generated. The same reason there's no need to declare/define copy/move constructors, etc, which haven't been done here, and are also being generated implicitly.

https://en.cppreference.com/w/cpp/language/destructor#Implicitly-declared_destructor

Reply:
is it considered bad form? Any reason to remove it?

Reply:
I think it should not be written out for the same reason all the other implicitly-generated functions aren't advised to be explicitly written out: less code = easier to read

Reply:
removed